tasks/libs/common/gomodules.py

```diff
@@ -0,0 +1,344 @@
+    base_dir: Path
+    # All GoModule to be taken into account (module.path: module)
+    modules: dict[str, GoModule]
+    # Name of each ignored module (not within `modules`)
+    ignored_modules: set[str]
+
+    @staticmethod
+    def from_dict(data: dict[str, dict[str, object]], base_dir: Path | None = None) -> Configuration:
+        base_dir = base_dir or Path.cwd()
+
+        modules = {}
+        ignored_modules = set()
+
+        for name, module_data in data.get('modules', {}).items():
+            if module_data == 'ignored':
+                ignored_modules.add(name)
+            elif module_data == 'default':
+                modules[name] = GoModule.from_dict(name, {})
+            else:
+                modules[name] = GoModule.from_dict(name, module_data)
+
+        return Configuration(base_dir, modules, ignored_modules)
+
+    @classmethod
+    def from_file(cls, base_dir: Path | None = None) -> Configuration:
+        """Load the configuration from a yaml file."""
+
+        base_dir = base_dir or Path.cwd()
+
+        with open(base_dir / cls.FILE_NAME) as file:
+            data = yaml.safe_load(file)
+
+        return Configuration.from_dict(data)
+
+    def to_dict(self) -> dict[str, object]:
+        modules_config = {}
+        # Path removed because the key is the path
+        modules_config.update(
+            {name: module.to_dict(remove_path=True) or 'default' for name, module in self.modules.items()}
+        )
+        modules_config.update({module: 'ignored' for module in self.ignored_modules})
+
+        return {
+            'modules': modules_config,
+        }
+
+    def to_file(self):
+        """Save the configuration to a yaml file at <base_dir/FILE_NAME>."""
+
+        with open(self.base_dir / self.FILE_NAME, "w") as file:
+            path = f'tasks/{Path(__file__).relative_to(Path(tasks.__file__).parent).as_posix()}'
+            print(self.INFO_COMMENT.format(file=path).strip() + '\n', file=file)
+
+            yaml.dump(self.to_dict(), file, Dumper=ConfigDumper)
+
+
+@dataclass
+class GoModule:
+    """A Go module abstraction.
+
+    See:
+        Documentation can be found in <docs/dev/modules.md>.
+
+    Args:
+        test_targets: Directories to unit test.
+        should_test_condition: When to execute tests, must be a enumerated field of `GoModule.CONDITIONS`.
+        should_tag: Whether this module should be tagged or not.
+        importable: HACK: Workaround for modules that can be tested, but not imported (eg. gohai), because they define a main package A better solution would be to automatically detect if a module contains a main package, at the cost of spending some time parsing the module.
+        independent: Specifies whether this modules is supposed to exist independently of the datadog-agent module. If True, a check will run to ensure this is true.
+        lint_targets: Directories to lint.
+        used_by_otel: Whether the module is an otel dependency or not.
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Worth mentioning defaults for each condition